build/NuGetAdditionalFiles/RunCsc

```diff
@@ -1,5 +1,15 @@
 #!/bin/sh
 
+set -e
+set -u
+
 THISDIR=$(dirname $0)
 
-dotnet $THISDIR/csc.dll "$@"
+# The expression ${VAR:+TEXT} expands to:
+#  If VAR is null or empty => empty string
+#  If VAR is non-empty => TEXT
+# Do this expression to make sure HOST_PATH ends with a slash,
+#  but if it's unset, have it be completely empty (look up dotnet in PATH)
+HOST_PATH=${DOTNET_HOST_PATH:+${DOTNET_HOST_PATH-}/}
```

Comment:
Also, the reason there's a dash at the end of `${DOTNET_HOST_PATH-}` in the inner expression is because I wasn't sure of evaluation order (outer->inner or inner->outer). If the inner is evaluated first, and `DOTNET_HOST_PATH` is unset, then that would trigger an error (if you have sane bash options set). However, as we *know* it can possibly be unset in non-error-cases, adding the `-` does a simple "replace with empty string if unset" instead of erroring. (`${VAR-TXT}`/`${VAR:-TXT}` are "if `VAR` set, return `${VAR}`, else return `TXT`, with `:` behavior differing on if `VAR` is the empty string - when the replacement string *is* the empty string, though, it doesn't matter, so `-` is fine)

tl;dr variables are complicated